### ISSUE TITLE
Covenant API

### ISSUE THREAD

**Issue by danjagnow** [blizzard-net/warcraft#159]
Add support for the Covenant API, which is a new addition to the [Game Data APIs](https://develop.battle.net/documentation/world-of-warcraft/game-data-apis) for the Shadowlands expansion.  See [WoW Shadowlands API Update - Covenants, Soulbinds, & More](https://us.forums.blizzard.com/en/blizzard/t/wow-shadowlands-api-update-covenants-soulbinds-more/13385) for details about the new APIs.

**Comment by willwolfram18** [comment]
I'll take a swing at this. Seems like a small and easy enough API set.

**Comment by willwolfram18** [comment]
@danjagnow wanted to check and see if you'd be cool with some tweaks I was interested in introducing while unit testing this method. Specifically I was looking at doing the following:

* Including [Fluent Assertions](https://fluentassertions.com/) for some validation of model deserializations
* Tests for different region+locale combos to better exercise some of the other code paths beyond the `Region.US` and `Locale.en_US`
    * Not planning to do all combos, but at least get a little more coverage.
* Change up the test case layout for the Covenant API methods, similar to the [DI tests](https://github.com/blizzard-net/warcraft/blob/master/tests/ArgentPonyWarcraftClient.Extensions.DependencyInjection.Tests/ServiceCollectionExtensionsTests/When_Adding_Warcraft_Clients.cs)
    * Basically looking at trying to bring a little SRP to the unit tests so that 1 class focuses on testing one method (or in this case pair of methods for the region+locale overload).

Let me know your thoughts, don't want to add too much or create fragmentation if you're not on-board.

**Comment by danjagnow** [comment]
- Adding **Fluent Assertions** would be good.  I use it in my professional work, and I like it.  I have just been too lazy to add validations beyond the not-null check.  This is particularly important after [the PR that migrated from Newtonsoft.Json to System.Text.Json](https://github.com/blizzard-net/warcraft/pull/142).  Previously it was possible to set `MissingMemberHandling = MissingMemberHandling.Error` for debug builds, which made it much easier to discover cases where the C# models lacked a property that was needed for full deserialization of the JSON.  See [the note about MissingMemberHandling in the migration guidance](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-5-0#missingmemberhandling) for details.  Hopefully https://github.com/dotnet/runtime/issues/37483 will get addressed in the future.
- Tests for different combinations of region and locale would be very welcome; again, I've just been too lazy to add these.
- Changing the test case layout is fine with me.  I find that the existing layout is fine for the level of testing that is currently being performed, but having a single test class per API may be too coarse-grained when adding the variants you're proposing.  Feel free to change it up.
